src/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -103,6 +103,7 @@ private void InitializeUri(ParsingError err, UriKind uriKind, out UriFormatExcep
                             _syntax = null; // convert to relative uri
                             e = null;
                             _flags &= Flags.UserEscaped; // the only flag that makes sense for a relative uri
+                            return;
```

Comment:
The return on this line is bypassing the IRI rule parsing which would ensure that the rest of the URI is valid (the `if` on line 119).
While not obvious, the Uri object is actually implementing the IRI RFCs since .Net 4.5. This change may allow invalid IRIs to pass through.

We probably need more testing that includes RFC 3987 relative IRIs (section 6.5) with various Unicode characters (e.g. RTL, reserved, pct-encoded, etc).